ocaml/stdlib/array.ml

```diff
@@ -23,7 +23,7 @@ type 'a t = 'a array
 
 (* Array operations *)
 
-external length : 'a array -> int = "%array_length"
+external length : 'a array @ contended -> int = "%array_length"
```

Comment:
`a -> b` without modes means legacy modes, which is `uncontended`. My mental model of `length` is that it only reads  an immutable section in the metadata of the array, hence doesn't require `uncontended` access to the array.

> Are there other similar functions that should change?

There might - but it's sound to forget that. We can lazily fix those when needed. Most data types cross `contention` anyway.

Reply:
I'd rather not touch the stdlib here. If a test needs this, then I would just redefine it in the context of the test.

Reply:
May I ask the conservation about stdlib? Note that we already modified stdlib with `[@local_opt]` anyway. Also with the removal of the upstream build, it matters even less now. So it's nice to ship with the compiler a stdlib that's "enriched".

Reply:
I can think of a few reasons to leave this alone:

1. It could potentially pose problems for code that must work both within Jane Street and outside of JS, including stuff we open-source, and stuff we import from the outside world. The two stdlibs will be different, and so it's conceivable that code that works in one place will fail in the other.
2. Updating just a few, arbitrary functions in the stdlib to have modes isn't actually helpful. If e.g. Base can't rely on a fully updated stdlib, the folks updating Base will have to consult each function individually, anyway.
3. It makes upstreaming harder. I expect a nice debate to crop up when we start upstreaming this stuff -- especially around libraries, where the design space is much larger. By not touching the stdlib (at all), we can have the debate about language features separate from library features.

So, mostly I see costs to making changes to the stdlib, while the only benefit is that a few tests can avoid writing some `external`s.

An alternative I would be happy with is a module in the right otherlib (by "right", I mean the otherlib that corresponds to the relevant extension level, which for this is `stable`) which could shadow the stdlib module and is fully updated for modes. Maybe it would be OK even not fully updated. But our new otherlibs situation allows us to have functions in the compiler that work with our new features without touching the stdlib.

Reply:
OK if we already think `otherlibs` is the place for enriched libraries, then I shouldn't touch stdlib.
I will leave the work in `otherlibs` to future PRs.

Reply:
I will also remove those tests that relies on the changed stdlib, because they are trivial, and not worth having `external` for.